File: MileageTracker/src/com/switkows/mileage/Charts/MileageChart.java
package com.switkows.mileage.Charts;

import java.util.ArrayList;
import java.util.List;
import org.achartengine.chart.*;

import com.switkows.mileage.MileageData;

import android.content.Context;
import android.graphics.Color;

public class MileageChart extends TimeChartExtension {
   // strings which will return series index
   public static final int           COMPUTER_MILEAGE = 0, ACTUAL_MILEAGE = 1;

   private static final String[]     mTitles          = {"MPG over Time", "Km/L over Time"};
   private static final String[]     mUnits           = {"MPG", "Km/L"};
   private static final String[]     mLineTitles      = {"Computer Mileage", "Actual Mileage"};
   private static final int[]        colors           = {Color.BLUE, Color.RED};
   private static final PointStyle[] styles           = {PointStyle.CIRCLE, PointStyle.SQUARE};

   public MileageChart(Context c, MileageData[] data, boolean isUS) {
      super(c, mTitles[isUS ? 0 : 1], mUnits[isUS ? 0 : 1], mLineTitles, colors, styles, data);
   }

   @Override
   protected void appendDataToSeries(long date, float[] values) {
      appendDataToSeries(date, COMPUTER_MILEAGE, values[0]);
      appendDataToSeries(date, ACTUAL_MILEAGE, values[1]);
   }

   @Override
   protected List<double[]> buildValuesList(MileageData[] data) {
      double[] comp_mpg = new double[data.length];
      double[] act_mpg = new double[data.length];
      List<double[]> values = new ArrayList<double[]>();
      for(int i = 0; i < mTitles.length; i++) {
         for(int row = 0; row < data.length; row++) {
            comp_mpg[row] = data[row].getComputerMileage(mContext, null);
            act_mpg[row] = data[row].getActualMileage(mContext, null);
         }
      }
      values.add(comp_mpg);
      values.add(act_mpg);
      return values;
   }
}

File: app/src/main/java/com/switkows/mileage/Charts/MileageChart.kt
package com.switkows.mileage.Charts

import java.util.ArrayList
import org.achartengine.chart.*

import com.switkows.mileage.MileageData

import android.content.Context
import android.graphics.Color

class MileageChart(c: Context, data: Array<MileageData>, isUS: Boolean) : TimeChartExtension(c, mTitles[if (isUS) 0 else 1], mUnits[if (isUS) 0 else 1], mLineTitles, colors, styles, data) {

    override fun appendDataToSeries(date: Long, values: FloatArray) {
        appendDataToSeries(date, COMPUTER_MILEAGE, values[0])
        appendDataToSeries(date, ACTUAL_MILEAGE, values[1])
    }

    override fun buildValuesList(data: Array<MileageData>): List<DoubleArray> {
        val comp_mpg = DoubleArray(data.size)
        val act_mpg = DoubleArray(data.size)
        val values = ArrayList<DoubleArray>()
        mTitles?.forEach { _ ->
            for (row in data.indices) {
                comp_mpg[row] = data[row].getComputerMileage(mContext, null).toDouble()
                act_mpg[row] = data[row].getActualMileage(mContext, null).toDouble()
            }
        }
        values.add(comp_mpg)
        values.add(act_mpg)
        return values
    }

    companion object {
        // strings which will return series index
        private val COMPUTER_MILEAGE = 0
        private val ACTUAL_MILEAGE = 1

        private val mTitles = arrayOf("MPG over Time", "Km/L over Time")
        private val mUnits = arrayOf("MPG", "Km/L")
        private val mLineTitles = arrayOf("Computer Mileage", "Actual Mileage")
        private val colors = intArrayOf(Color.BLUE, Color.RED)
        private val styles = arrayOf(PointStyle.CIRCLE, PointStyle.SQUARE)
    }
}
